File: algorithmsJS/CountLetters.js
class CountLetters {
  static countLetters(s) {
    let result = "";
    let count = 1;
    for (let i = 1; i < s.length; i++) {
      if (s.charAt(i) == s.charAt(i - 1)) {
        count++;
      } else {
        result = result + s.charAt(i - 1) + count;
        count = 1;
        continue;
      }
      if (i + 1 == s.length) {
        result = result + s.charAt(i - 1) + count;
        count = 1;
      }
    }
    return result;
  }

  static countLetters1(s) {
    return 0;
  }
}

// Example usage:
const input = "aaaabbccaaaaa";
const result = CountLetters.countLetters(input);
const result1 = CountLetters.countLetters1(input);
console.log(`Count repeating characters for input "${input}": ${result}`);
console.log(`Count repeating characters for input "${input}": ${result1}`);

File: algorithms/CountLetters.java
package algorithms;

public class CountLetters {

    // Method to count letters
    public static String countLetters(String s) {
        if (s == null || s.length() == 0)
            return "";

        StringBuilder result = new StringBuilder();
        int count = 1;

        for (int i = 1; i < s.length(); i++) {
            if (s.charAt(i) == s.charAt(i - 1)) {
                count++;
            } else {
                result.append(s.charAt(i - 1)).append(count);
                count = 1;
            }

            if (i + 1 == s.length()) {
                result.append(s.charAt(i)).append(count);
            }
        }

        return result.toString();
    }

    // Placeholder like your countLetters1
    public static int countLetters1(String s) {
        return 0;
    }

    // Example usage
    public static void main(String[] args) {
        String input = "aaaabbccaaaaa";
        String result = countLetters(input);
        int result1 = countLetters1(input);

        System.out.println("Count repeating characters for input \"" + input + "\": " + result);
        System.out.println("Count repeating characters for input \"" + input + "\": " + result1);
    }
}
